recipes/msys2/all/conanfile.py

```diff
@@ -11,54 +42,175 @@ class MSYS2Conan(ConanFile):
     homepage = "http://www.msys2.org"
     license = "MSYS license"
     topics = ("conan", "msys", "unix", "subsystem")
-    build_requires = "7zip/19.00"
     short_paths = True
-    options = {"exclude_files": "ANY",  # Comma separated list of file patterns to exclude from the package
-               "packages": "ANY",  # Comma separated
-               "additional_packages": "ANY"}    # Comma separated
-    default_options = {"exclude_files": "*/link.exe",
-                       "packages": "base-devel,binutils,gcc",
-                       "additional_packages": None}
-    settings = "os_build", "arch_build"
-
-    def configure(self):
-        if self.settings.os_build != "Windows":
+    # "exclude_files" "packages" "additional_packages" values are a comma separated list
+    options = {
+        "exclude_files": "ANY",
+        "packages": "ANY",
+        "additional_packages": "ANY"
+    }
+    default_options = {
+        "exclude_files": "*/link.exe",
+        "packages": "base-devel,binutils,gcc",
+        "additional_packages": None
+    }
+    settings = "os", "arch"
+
+
+    def validate(self):
+        if self.settings.os != "Windows":
             raise ConanInvalidConfiguration("Only Windows supported")
-
-    def source(self):
-        # build tools have to download files in build method when the
-        # source files downloaded will be different based on architecture or OS
-        pass
+        if tools.Version(self.version) >= "20210105" and self.settings.arch != "x86_64":
+            raise ConanInvalidConfiguration("Only Windows x64 supported")
+        if tools.Version(self.version) <= "20161025":
+            raise ConanInvalidConfiguration("msys2 v.20161025 is no longer supported")    
```

Comment:
Is this required since the version was removed?

Reply:
if you don't remove it, you will receive `zero packages generated` error for version 20161025 
but right now we're not sure these versions make sense at all, as we upgrade them right away, so effectively all the versions are the same

Reply:
Actually we are waiting for a decision regarding rolling releases. :-) cc @jgsogo